extensions/amp-sidebar/0.1/amp-sidebar.js

```diff
@@ -0,0 +1,254 @@
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {CSS} from '../../../build/amp-sidebar-0.1.css';
+import {Layout} from '../../../src/layout';
+import {assert} from '../../../src/asserts';
+import {isExperimentOn} from '../../../src/experiments';
+import {dev} from '../../../src/log';
+import {platform} from '../../../src/platform';
+import {setStyles} from '../../../src/style';
+
+
+/** @const */
+const EXPERIMENT = 'amp-sidebar';
+
+/** @const */
+const TAG = 'AmpSidebar';
+
+/** @const */
+const WHITELIST_ = ['AMP-ACCORDION', 'AMP-FIT-TEXT', 'AMP-IMG'];
+
+/** @const */
+const IOS_SAFARI_BOTTOMBAR_HEIGHT = '10vh';
+
+export class AmpSidebar extends AMP.BaseElement {
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return layout == Layout.CONTAINER;
+  }
+
+  /** @override */
+  isReadyToBuild() {
+    return false;
+  }
+
+  /** @override */
+  buildCallback() {
+    /** @const @private {boolean} */
+    this.isExperimentOn_ = isExperimentOn(this.getWin(), EXPERIMENT);
+
+    /** @private @const {!Window} */
+    this.win_ = this.getWin();
+
+    /** @private @const {!Document} */
+    this.document_ = this.win_.document;
+
+    /** @private @const {!Element} */
+    this.documentElement_ = this.document_.documentElement;
+
+    /** @private @const {string} */
+    this.direction_ = this.element.getAttribute('direction');
+
+    /** @private @const {!Viewport} */
+    this.viewport_ = this.getViewport();
+
+    /** @private @const {boolean} */
+    this.hasMask_ = false;
+
+    /** @private @const {boolean} */
+    this.isIosSafari_ = platform.isIos() && platform.isSafari();
+
+    if (this.direction_ != 'left' && this.direction_ != 'right') {
+      const pageDir = this.document_.body.getAttribute('dir') || 'ltr';
+      this.direction_ = (pageDir == 'rtl') ? 'right' : 'left';
+      this.element.setAttribute('direction', this.direction_);
+    }
+
+    if (!this.isExperimentOn_) {
+      dev.warn(TAG, `Experiment ${EXPERIMENT} disabled`);
+      return;
+    }
+    if (!this.checkWhitelist_()) {
+      return;
+    }
+
+    if (this.isIosSafari_) {
+      this.fixIosElasticScrollLeak_();
+    }
+
+    this.adjustPadding_();
+
+    if (this.documentElement_.classList.contains('amp-sidebar-open')) {
+      // Create the mask if the sidebar is rendered in open mode.
+      this.createMask_();
+    }
+
+    this.documentElement_.addEventListener('keydown', event => {
+      // Close sidebar on ESC.
+      if (event.keyCode == 27) {
+        this.close_();
+      }
+    });
+    //TODO (skrish, #2712) Add history support on back button.
+    this.registerAction('activate', this.open_.bind(this));
```

Comment:
This won't work. Specifically with activate. Because BaseElement already defines `activate` method. Instead, you need to simply override it and call `open_` from it.